traits/src/data_provider.rs

```diff
@@ -2,9 +2,9 @@ use sp_runtime::DispatchResult;
 use sp_std::vec::Vec;
 
 /// Data provider with ability to provide data with no-op, and provide all data.
-pub trait DataFeeder<Key, Value, AccountId>: DataProvider<Key, Value> {
+pub trait DataFeeder<Key, Value, AccountId> {
```

Comment:
I think not having the dependency on `DataProvider` allows entities to implement only the feeder part, which can be helpful. This should not be an API change, but please feel free to tell me if you prefer not, and I revert it.

Reply:
that's fine